### DISCUSSION TITLE
Prepare for version 1.0.0, please

### DISCUSSION THREAD

**Discussion by kasparkallas** [wevm/wagmi#1603] [Ideas]
`wagmi` is being used in production dApps and libraries. It's a great library and my first choice when starting a new dApp project or when I need to recommend wallet connecting library to others. It clearly has great adoption when looking at NPM trends: https://npmtrends.com/@usedapp/core-vs-wagmi-vs-web3-react-vs-web3modal

Based on the success, can you prepare for a stable version 1.0.0 release soon, please? 

NPM doesn't work well with 0.x.x versions. For example, if I have specified `^0.x.y` then it will only update `y` but not `x` like you would expect when you have `1.x.y`. This is probably what you guys want as I see you've been making big structural and breaking changes lately (ES modules only, separating chains & connectors packages, etc). https://semver.org/#how-should-i-deal-with-revisions-in-the-0yz-initial-development-phase

But as a consumer of `wagmi`, I use it along with RainbowKit and Gnosis Safe connector, and I'm currently blocked from updating to the latest version of `wagmi` because the peer dependency version specifications are all out of sync for these libraries at the time of writing. This might all be part of the plan of current fast `wagmi` development... but as a consumer and a fan, I'm not too fond of it.

**Comment by tmm** [comment]
Appreciate the feedback!

wagmi will hit 1.0.0 sometime in the new year and follow semver more closely. Until then, wagmi is in a production stable beta with breaking changes released in minor versions (e.g. 0.x.0). This allows us to iterate more quickly on the API design ahead of the first major version release. We publish a [migration guide](https://wagmi.sh/react/migration-guide) that details breaking changes between minor versions and are usually in discussion with ecosystem projects (e.g. RainbowKit) ahead of releases to give them time to update (whether or not they do that is up to them).
